compiler/test/dotty/tools/repl/LoadTests.scala

```diff
@@ -0,0 +1,75 @@
+package dotty.tools.repl
+
+import java.nio.file.{ Path, Files }
+
+import org.junit.{ Test, BeforeClass }
+import org.junit.Assert.assertEquals
+
+class LoadTests extends ReplTest {
+  import LoadTests._
+
+  @Test def helloworld = loadTest(
+    file    = """def helloWorld = "Hello, World!"""",
+    defs    = """|def helloWorld: String
+                 |
+                 |
+                 |""".stripMargin,
+    runCode = "helloWorld",
+    output  = """|val res0: String = Hello, World!
+                 |
+                 |""".stripMargin
+  )
+
+  @Test def maindef = loadTest(
+    file    = """@main def helloWorld = println("Hello, World!")""",
+    defs    = """|def helloWorld: Unit
+                 |
+                 |
+                 |""".stripMargin,
+    runCode = "helloWorld",
+    output  = """|Hello, World!
+                 |
+                 |""".stripMargin
+  )
+
+  @Test def maindefs = loadTest(
+    file    = """|@main def helloWorld = println("Hello, World!")
+                 |@main def helloTo(name: String) = println(s"Hello, $name!")""".stripMargin,
+    defs    = """|def helloTo(name: String): Unit
+                 |def helloWorld: Unit
+                 |
+                 |
+                 |""".stripMargin,
+    runCode = """helloWorld; helloTo("Scala")""",
+    output  = """|Hello, World!
+                 |Hello, Scala!
+                 |
+                 |""".stripMargin
+  )
+
+  def loadTest(file: String, defs: String, runCode: String, output: String) =
+    eval(s":load ${writeFile(file)}").andThen { implicit s =>
+      assertEquals(defs, storedOutput())
+      run(runCode)
+      assertEquals(output, storedOutput())
+    }
+
+  private def eval(code: String): State =
+    fromInitialState { implicit s => run(code) }
+
+}
+
+object LoadTests {
+
+  var dir: Path = null
+
+  @BeforeClass def setupDir: Unit =
+    dir = Files.createTempDirectory("repl_load_src")
+
```

Comment:
We should probably delete the directory `@AfterClass` just to be sure.

Reply:
I've added this suggestion, thanks